src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -385,11 +387,11 @@ public DateTime Add(TimeSpan value)
 
         // Returns the DateTime resulting from adding a fractional number of
         // time units to this DateTime.
-        private DateTime Add(double value, int scale)
+        private static DateTime Add(DateTime date, double value, int scale)
```

Comment:
We explicitly did this for the various hardware intrinsic methods specifically because it made it easier to optimize and improved codegen.

They are admittedly a bit more perf sensitive than DateTime may be however 😄

This could always be an extension method in which case it can still be consumed the same while still avoiding `date` being an implicit `byref`